swri_roscpp/readme.md

```diff
@@ -31,3 +31,51 @@ add_topic_service_files(DIRECTORY topic_srv FILES
 ```
 
 To use the generated topics simply include `your_project/your_service.h`. They have response and request fields just like normal services.
+
+## Dynamic Parameters
+
+A class that implements the same functionality as Dynamic Reconfigure, but in a more dynamic and easier to use way. Instead of declaring configurable parameters at compile time in a config file they are specified in your C++ code.
+
+To use, add the class to your node and call the initalize function with a NodeHandle in your private namespace.
+
+```cpp
+#include <swri_roscpp/dynamic_parameters.h>
+
+void main()
+{
+  // your node stuff here
+
+  ros::NodeHandle pnh("~");
+  swri::DynamicParameters params;
+  params.initialize(pnh);
+...
+```
+
+You can then start declaring and reading in configuration values, giving a long living pointer to a variable for each. This variable is changed on a dynamic reconfigure of that parameter. If you want to ensure no changes are applied during an operation, be sure to lock the mutex you can get by calling the `params.mutex()` function.
+
+```cpp
+  float flt;
```

Comment:
Setting the params like this is actually very dangerous.  The DynamicParameters class is storing a reference to this variable, but this variable will be destroyed when the function calling it goes out of scope.  The next time your parameters are reconfigured after that, it will cause a segfault.

I think it'd be better to make wrapper types (`BoolParam`, `IntParam`, etc.) that are actually `shared_ptr`s under the hood in order to make sure they always exist; it would also be possible to instrument those types with getters that could handle automatically locking them to make sure they're thread-safe.

Reply:
@pjreed went and added something like this